software/lazydogpile.livecodescript

```diff
@@ -0,0 +1,954 @@
+   refreshDraw
+   refreshDisplay
+end hideButtMouseUp
+
+on tooltipSwitchChange
+   put the hilite of button "tooltipSwitch" stack "howstrBeta" into tX
+   put tX into sTooltipVisible
+end tooltipSwitchChange
+
+on busySwitchChange
+   get the hilite of button "busySwitch" stack "howstrBeta"
+   --   put it into sBusy
+   set the cBusy of stack "HowstrBeta" to it
+end busySwitchChange
+
+on caseButtMouseUp
+   -- load up existing case node title if it exists
+   put sMainShort[sNowDoc][2]["title"] into tCurrent
+   if tCurrent is empty then
+      ask "name for case node"
+   else
+      ask "name for case node" with tCurrent
+   end if
+   -- only request a change if the case node title actually changed
+   if it is not tCurrent then
+      put it into tStuff["data"]
+      put "title" into tStuff["tag"]
+      put sNowDoc into tStuff["doc"]
+      put 2 into tStuff["rec"]
+      changeInfo sMain,tStuff -- grapheriffic
+      refreshDraw
+      refreshDisplay
+   end if
+end caseButtMouseUp
+
+on miscRightButtMouseUp
+   -- slide all of the misc nodes one place to the right
+   repeat for each key tRec in sGUIstuff["nodes"]
+      if sGUIstuff["nodes"][tRec]["misc"] is true then 
+         put sGUIstuff["nodes"][tRec]["horz"] into tH
+         put sGUIstuff["nodes"][tRec]["vert"] into tV
+         put tH + sNodeSize + sLayerSize into tH
+         set the loc of sGUIstuff["nodes"][tRec]["graphic"] to tH & "," & tV
+         put tH into sGUIstuff["nodes"][tRec]["horz"]
+      end if
+   end repeat
+end miscRightButtMouseUp
+
+on miscLeftButtMouseUp
+   -- slide all of the misc nodes one place to the left
+   repeat for each key tRec in sGUIstuff["nodes"]
+      if sGUIstuff["nodes"][tRec]["misc"] is true then 
+         put sGUIstuff["nodes"][tRec]["horz"] into tH
+         put sGUIstuff["nodes"][tRec]["vert"] into tV
+         put tH - sNodeSize - sLayerSize into tH
+         set the loc of sGUIstuff["nodes"][tRec]["graphic"] to tH & "," & tV
+         put tH into sGUIstuff["nodes"][tRec]["horz"]
+      end if
+   end repeat
+end miscLeftButtMouseUp
+
+on diveButtMouseUp
+   -- dive from the case node to the focused node
+   -- new elements are identified by letters (a, b, c, etc); existing elements id'd by their record #
+   -- x[letter][tag]
+   -- x[ltr][type]=flow/dive/rise/state/change
+   -- x[ltr][backward/forward]=ltr/#
+   put "dive" into tDo["a"]["type"]
+   put sGUIstuff["case"] into tDo["a"]["backward"]
+   put sMouseHistory["nodeSelected"] into tDo["a"]["forward"]
+   changeRequest tDo,sMain,sNowDoc
+   refreshDraw
+   refreshDisplay
+end diveButtMouseUp
+
+on riseButtMouseUp
+   -- rise from the focused node to the case node
+   put "rise" into tDo["a"]["type"]
+   put sGUIstuff["case"] into tDo["a"]["forward"]
+   put sMouseHistory["nodeSelected"] into tDo["a"]["backward"]
+   changeRequest tDo,sMain,sNowDoc
+   refreshDraw
+   refreshDisplay
+end riseButtMouseUp
+
+on zoomInButtMouseUp
+   -- increase dimensions by 10%
```

Comment:
@matthewmaier I'm not sure if it matters in this instance but this method will end up with rounding errors so scaling up and then back down might not end up with exactly the same dimensions. If you want to avoid it you will need to scale on the formatted width and height of the image and then keep track of the current scale factor.